src/SonarScanner.MSBuild.PreProcessor/Roslyn/RoslynAnalyzerProvider.cs

```diff
@@ -28,28 +28,26 @@
 
 namespace SonarScanner.MSBuild.PreProcessor.Roslyn;
 
-public class RoslynAnalyzerProvider : IAnalyzerProvider
+public class RoslynAnalyzerProvider(IAnalyzerInstaller analyzerInstaller, ILogger logger) : IAnalyzerProvider
```

Comment:
This is too hard to review. See https://xtranet-sonarsource.atlassian.net/wiki/spaces/LANG/pages/3790045334/Pull+request+creation+guidelines
I've asked you to create PR with cleanup, and then PR with the actual change.

The cleanup should redesign the structure of this spaghetti code first, so the 2nd PR would be easy to implement. 

Please split it.